### PR TITLE
Tdd project

### PR DESCRIPTION
**Pull request by TEMESGEN-G25**
## Summary

- Create a class called `Solver`.
- Create a method called `factorial` that takes one argument, an integer `N`, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to `N` and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Create a method called `reverse` that takes one argument, a string `word`, and returns `word` reversed (e.g. if `word` is `"hello"` it returns `"olleh"`).
- Create a method called `fizzbuzz` that takes one argument, an integer `N`, and returns a string. The returned string is constructed following these rules:
  - When `N` is divisible by 3, return `"fizz"`.
  - When `N` is divisible by 5, return `"buzz"`.
  - When `N` is divisible by 3 and 5, return `"fizzbuzz"`.
  - Any other case, return `N` as a string (e.g. say `N` is `7` then return `"7"`).
- Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,5 +1,3 @@
 source 'https://rubygems.org'
```
Comment (Ismailco) [inline Gemfile]: - [x] [OPTIONAL] You should Use RSpec for unit testing, kindly add it to the gemfile so others can use your code out of the box

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+  it 'Create new instance of solver class' do
+    @solver.should be_an_instance_of Solver
+  end
+
+  it 'Return the factorial of the number without breaking' do
```
Comment (Ismailco) [inline spec/solver_spec.rb]: - [x] Your tests for the factorial method should also include a test for Negative value which raises an exception as an expectation

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @NatiG25, @bizip, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile (hunk 1 above)
Comment (TEMESGEN-G25) [reply Gemfile]: @Ismailco Thank you for your suggestions. So you mean to simply write these two lines inside the file:
gem 'rubocop', '>= 1.0', '< 2.0'
gem "rspec"

Comment (Ismailco) [reply Gemfile]: @NatiG25, Yes

File: spec/solver_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    it 'Return the factorial of the number without breaking' do
+      expect(@solver.factorial(4)).to eql 24
+    end
+
+    it 'Return 1 if the input is 0' do
+      expect(@solver.factorial(0)).to eql 1
+    end
+
+    it 'Raises an exception' do
+      # expect(@solver.factorial(-9)).to eql
```
Comment (Ismailco) [inline spec/solver_spec.rb]: - [x] [OPTIONAL] Kindly remove any unused commented code from your repo, it makes your code messy

**Review by Ismailco — APPROVED** [review]
### Hi @NatiG25, @bizip, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
